### PR TITLE
[DependencyInjection] Allow `Target` attribute to be used with constructor property promotion

### PR DESCRIPTION
**Pull request by ruudk**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      |no
| New feature?  | yes
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

This makes it possible to do this:
```php
class GitHubDownloader
{
    public function __construct(#[Target('githubApi')] private HttpClientInterface $httpClient)
    {
    }
}
```

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Attribute/Target.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
  *
  * @author Nicolas Grekas <p@tchwork.com>
  */
-#[\Attribute(\Attribute::TARGET_PARAMETER)]
+#[\Attribute(\Attribute::TARGET_PARAMETER | \Attribute::TARGET_PROPERTY)]
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Attribute/Target.php:21]: I'm not sure this PR is needed, this works already to me, see eg https://3v4l.org/W0kFq

**Comment by ruudk** [comment]
Seems like a PHPStorm issue :(  
Screenshot 2021-07-10 at 19 21 27@2x
It does work indeed.

File: src/Symfony/Component/DependencyInjection/Attribute/Target.php (hunk 1 above)
Comment (ruudk) [reply src/Symfony/Component/DependencyInjection/Attribute/Target.php:21]: Seems like a PHPStorm issue then :( 
Screenshot 2021-07-10 at 19 21 27@2x

**Comment by AndreasA** [comment]
@nicolas-grekas I know this is a bit older but I just noticed this myself in PhpStorm and I did some more analysis and though it works, the attribute actually gets applied to the property and the constructor parameter in this case which is why PhpStorm decided to mark this as an error/warning.

see https://3v4l.org/NvcWA
and corresponding issue https://youtrack.jetbrains.com/issue/WI-59609/Non-applicable-attribute-target-declaration-false-positive-for-Attributes-and-Constructor-property-promotion and the linked discussion in it https://externals.io/message/111942

so I think it would be more correct to add property as target as well. though it might be confusing for users if it is applied directly to a property then.

**Comment by ruudk** [comment]
Yes, this is extremely annoying from a developer perspective.

**Comment by stof** [comment]
please open new issues if you want to discuss that instead of commenting on a PR closed since years.

**Pull request closed without merging**